src/Init/Core.lean

```diff
@@ -1684,15 +1684,15 @@ axiom ofReduceNat (a b : Nat) (h : reduceNat a = b) : a = b
 `IsAssociative op` says that `op` is an associative operation,
 i.e. `(a ∘ b) ∘ c = a ∘ (b ∘ c)`. It is used by the `ac_rfl` tactic.
 -/
-class IsAssociative {α : Sort u} (op : α → α → α) where
+class IsAssociative {α : Sort u} (op : α → α → α) : Type where
```

Comment:
IMO these should be Props, this is exactly the kind of issue we are hoping to avoid by changing the default